src/Components/reusables/pro-upsell/ProUpsellDisplay.tsx

```diff
@@ -71,12 +80,63 @@ const ProUpsellDisplay: FunctionComponent<ProUpsellDisplayProps> = ({
   closeAction,
   closeText,
 }) => {
+  const [proPackage, setPackage] = React.useState<PurchasesOffering | null>();
+  const [purchaseError, setPurchaseError] = React.useState<string | null>();
+  const [loading, setLoading] = React.useState(false);
+  const dispatch = useAppDispatch();
+
   React.useEffect(() => {
     sendEvent('page_view', {
       screen: 'pro_upsell',
     });
+
+    fetchOfferings();
   }, []);
 
+  const fetchOfferings = async () => {
+    const offerings: PurchasesOfferings = await Purchases.getOfferings();
+    if (
+      offerings.current !== null &&
+      offerings.current.availablePackages.length !== 0
+    ) {
+      setPackage(offerings.current);
+    }
+  };
+
+  const purchasePackage = async () => {
+    try {
+      setLoading(true);
+
+      sendEvent('pro__click', {
+        screen: 'pro_upsell',
+      });
+
+      await Purchases.purchasePackage(
+        proPackage?.availablePackages[0] as PurchasesPackage,
+      );
+
+      sendEvent('pro__register', {
```

Comment:
this event should only be sent if the user successfully registers

Reply:
Yeah, that's how it should work. If the register isn't successful, execution should stop there and go to the catch block.